sympy/polys/polytools.py

```diff
@@ -1470,7 +1470,11 @@ def mul(f, g):
         g = sympify(g)
 
         if not g.is_Poly:
-            return f.mul_ground(g)
+            try:
+                return f.mul_ground(g)
+            except CoercionFailed:
+                f = div(f, 1, domain = "QQ")[0]
+                return f.mul_ground(g)
```

Comment:
We shouldn't special case `QQ` here

Reply:
I believe case was called when there was a multiplication of a polynomial by a rational number (including integers). The problem arose because `mul_ground` tries to reduce the rational number to the domain type of the polynomial. When the polynomial was a rational , there was a coercion error because a non-integer rational number. So, setting the domain as QQ fixed this error.

I understand your concern, however. I checked `print(poly(((x+1)**2)/2**(1/2))` and it resulted in a very large fraction due to a fractional form of sqrt(2). Changing it to "RR" would umbrella that case as well. Do you believe that is a better idea?